dev/test/collection.ts

```diff
@@ -169,4 +175,51 @@ describe('Collection interface', () => {
     expect(coll1.isEqual(coll1Equals)).to.be.ok;
     expect(coll1.isEqual(coll2)).to.not.be.ok;
   });
+
+  it('for CollectionReference.withConverter()', async () => {
+    const doc = document('documentId', 'author', 'author', 'title', 'post');
+    const overrides: ApiOverride = {
+      commit: request => {
+        const expectedRequest = set({
+          document: doc,
+        });
+        requestEquals(request, expectedRequest);
+
+        return response(writeResult(1));
+      },
+      batchGetDocuments: () => {
+        const stream = through2.obj();
+        setImmediate(() => {
+          stream.push({found: doc, readTime: {seconds: 5, nanos: 6}});
+          stream.push(null);
+        });
+
+        return stream;
+      },
+    };
+
+    return createInstance(overrides).then(async firestore => {
+      const docRef = firestore
+        .collection('collectionId')
+        .withConverter(postConverter)
+        .doc('documentId');
+
+      await docRef.set(new Post('post', 'author'));
+      const postData = await docRef.get();
+      const post = postData.data();
+      expect(post).to.not.equal(undefined);
+      expect(post!.byline()).to.equal('post, by author');
+    });
+  });
+
+  it('drops the converter when calling CollectionReference<T>.parent()', () => {
+    return createInstance().then(async firestore => {
+      const postsCollection = firestore
+        .collection('users/user1/posts')
+        .withConverter(postConverter);
+
+      const usersCollection = postsCollection.parent;
+      expect(usersCollection!.isEqual(firestore.doc('users/user1'))).to.be.true;
```

Comment:
This tests relies on the fact that we compare converters in `isEqual`. You can keep the test as is, but you should change the semantics of `isEqual` so that the test would fail if we don't drop the converter.

Reply:
Added the `this.converter === other.converter` check for isEqual.